### PR TITLE
Added aggregate & conditonal readers for parquet

### PR DESCRIPTION
**Pull request by ajayborra**
**Related issues**
https://github.com/salesforce/TransmogrifAI/issues/56

**Describe the proposed solution**
Added Aggregate & Conditional Reader API for parquet format

### CODE REVIEW

**Comment by tovbinm** [comment]
@ajayborra please add some tests. thanks.

File: readers/src/main/scala/com/salesforce/op/readers/ParquetProductReader.scala [hunk 1]
```diff
@@ -56,3 +56,33 @@ class ParquetProductReader[T <: Product : Encoder]
...
+  * @param key      function for extracting key from record
+  * @param conditionalParams params for conditional aggregation
+  * @tparam T
+  */
+class ConditionalParquetProductReader[T <: Product : Encoder : WeakTypeTag]
+(
+  readPath: Option[String],
+  key: T => String,
+  val conditionalParams: ConditionalParams[T]
+)extends ParquetProductReader[T](readPath, key) with ConditionalDataReader[T]
```
Comment (tovbinm) [inline readers/src/main/scala/com/salesforce/op/readers/ParquetProductReader.scala]: please fix scalastyle warnings

Comment (ajayborra) [reply readers/src/main/scala/com/salesforce/op/readers/ParquetProductReader.scala]: @tovbinm Seems like we have are going to have [similar logic](https://github.com/salesforce/TransmogrifAI/blob/master/readers/src/test/scala/com/salesforce/op/readers/CSVProductReadersTest.scala#L119-L163) for testing these readers. Will look at the possibility of extracting these as Table Tests and update the PR.

Comment (vpatryshev) [reply readers/src/main/scala/com/salesforce/op/readers/ParquetProductReader.scala]: Any tests included? Why not?

**Comment by tovbinm** [comment]
@ajayborra Rachel is working on adding a common spec for all the readers, so dont sweat it too much ;) https://github.com/salesforce/TransmogrifAI/issues/83

**Pull request merged**